test/testDemoCmd.js

```diff
@@ -35,6 +36,33 @@ describe('Integration test for demo command', function suite() {
     await fs.remove(testDir);
   });
 
+  it('demo execAsync if/else branching correct', async () => {
+    const demoInstance = new DemoCommand();
+    assert.equal(0, await demoInstance.execAsync('git --version'));
+    assert.equal(127, await demoInstance.execAsync('falseCommandDummy').catch(() => 127));
+  });
+
+  it('test for resolve upon finding Git', async () => {
+    const demoInstance = new DemoCommand();
+    sinon.stub(demoInstance, 'execAsync').returns(0);
+
+    assert.doesNotReject(demoInstance
+      .withDirectory(testDir)
+      .withName('project1')
+      .withType('full')
+      .run());
+  });
+
+  it('test for failure, when Git is not installed', async () => {
+    const demoInstance = new DemoCommand();
+    sinon.stub(demoInstance, 'execAsync').returns(127);
```

Comment:
Instead of stubbing the `execAsync` command, it might be better to have a Mocha `before` and `after` that moves `.gitconfig` aside and then back after the test. It should be relatively easy to recreate the test condition and it will help us detect issues in case the `git` return status code changes, etc.